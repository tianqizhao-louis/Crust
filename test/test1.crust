/* 

    This is a test file
    For Crust
 */

int m;
int n;

/*
    The test is specifically created for helloworld
*/

int print_helloword(int m, int n) {
    if(m < n) {
        return m;
    }else {
        return n;
    }
}

// single-line comment, should be ignored

int main() {
  int x;
  int y;
  int z;

  m = 18;
  m = 9;
  x = 2;
  y = 14;

  print(print_helloword(x,y));
  print(print_helloword(3,15));

  int this_variable = 10;
  bool u = true;

  print(print_helloword(99, 121));
  print(print_helloword(m, n));
  return 0;
}
