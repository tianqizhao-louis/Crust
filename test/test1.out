

Sementically checked program: 

int m;
int n;

int print_helloword(m, n)
{
if ((bool : (int : m) < (int : n)))
{
return (int : m);
}
else
{
return (int : n);
}
}

int main()
{
int x;
int y;
int z;
int this_variable;
bool u;
(int : m = (int : 18));
(int : m = (int : 9));
(int : x = (int : 2));
(int : y = (int : 14));
(int : print((int : print_helloword((int : x), (int : y)))));
(int : print((int : print_helloword((int : 3), (int : 15)))));
(int : this_variable = (int : 10));
(bool : u = (bool : true));
(int : print((int : print_helloword((int : 99), (int : 121)))));
(int : print((int : print_helloword((int : m), (int : n)))));
return (int : 0);
}

